### DISCUSSION TITLE
Other Source Generators in Uno projects

### DISCUSSION THREAD

**Discussion by HavenDV** [unoplatform/uno#8175] [General]
I noticed that the Uno package started blocking the use of third-party generators. In my case, I was generating View code behind constructor code in my test project. 
Is this a bug or is it a special behavior to limit the possible problems of using other generators in most cases? In second case, just need to create a library that will call these generators and use it already in Uno

I assume it is caused by this code
https://github.com/unoplatform/uno/blob/2ed9559ff94e459938d227a4a309c4ba65e17056/src/SourceGenerators/Uno.UI.SourceGenerators/Content/Uno.UI.SourceGenerators.props#L362-L363

P.S. I did not find any changes in the code that would explicitly prohibit the use of other generators. But I still can't achieve using another generator in a project with Uno.UI.WebAssembly. This may be related to dotnet updates to 6.0.2 or VS to 17.2 preview

Repro:
[ratbuddyssey-master.zip](https://github.com/unoplatform/uno/files/8122613/ratbuddyssey-master.zip)
1. Build Ratbuddyssey.Apps.Uno.WebAssembly. The generator will not show the generated file (it should appear if at least one AdditionalFile exists at the time of generation)
<img width="230" alt="image" src="https://user-images.githubusercontent.com/3002068/155279466-e0162201-0725-4b4d-83e9-39c501759c22.png">
2. This is not a bug related to the generation order. If you remove Visible="False" then you can see that AdditionalFile exists before the start of build. But it seems then some MSBuild code removes AdditionalFile at one of the stages of preparation.
3. ViewBase code is generated in another library (Ratbuddyssey.Apps.Uno.Core.Standard) and available for Uno generator

**Comment by jeromelaban** [comment]
This particular behavior is not intentional, and it's not preventing other generators to run but rather removing unrelated additional files.

Let's open https://github.com/unoplatform/uno/issues/8176 to track this.
